package/src/prmon.cpp

```diff
@@ -12,71 +13,46 @@
 
 #include <cstddef>
 #include <fstream>
+#include <iomanip>
 #include <iostream>
-#include <unordered_map>
+#include <memory>
+#include <nlohmann/json.hpp>
 #include <sstream>
 #include <thread>
+#include <unordered_map>
 #include <vector>
-#include <iomanip>
-#include <memory>
-
-#include <nlohmann/json.hpp>
 
+#include "prmonutils.h"
 #include "registry.h"
-
-#include "pidutils.h"
-#include "prmon.h"
 #include "wallmon.h"
 
-bool sigusr1 = false;
-
-void SignalCallbackHandler(int /*signal*/) {
-  sigusr1 = true;
-}
-
-void SignalChildHandler(int /*signal*/) {
-  int status;
-  pid_t pid{1};
-  while (pid > 0) {
-    pid = waitpid((pid_t)-1, &status, WNOHANG);
-    if (status && pid > 0) {
-      if (WIFEXITED(status))
-        std::clog << "Child process " << pid
-                  << " had non-zero return value: " << WEXITSTATUS(status)
-                  << std::endl;
-      else if (WIFSIGNALED(status))
-        std::clog << "Child process " << pid << " exited from signal "
-                  << WTERMSIG(status) << std::endl;
-      else if (WIFSTOPPED(status))
-        std::clog << "Child process " << pid << " was stopped by signal"
-                  << WSTOPSIG(status) << std::endl;
-      else if (WIFCONTINUED(status))
-        std::clog << "Child process " << pid << " was continued" << std::endl;
-    }
-  }
-}
+bool prmon::sigusr1 = false;
 
 int MemoryMonitor(const pid_t mpid, const std::string filename,
```

Comment:
Maybe this might be a good time to finally convert this to `ProcessMonitor` instead of `MemoryMonitor`?

Reply:
Yes - ideal time to do that indeed. Done in last commit.